### ISSUE TITLE
Swagger Editor is sending my POST as a GET

### ISSUE THREAD

**Issue by webron** [swagger-api/swagger-editor#578]
_From @hzill on August 3, 2015 15:8_

I'm having trouble getting Swagger to send my request body as a raw JSON document using the "Try This Operation" feature. It keeps converting the JSON to a query string and sending the request as a GET, which my server rejects. I'm running locally, with the server on port 8080 and the editor on port 8081.

Here is my yaml:

```yaml
paths:
  /signup:
    post:
      description:
        signup a user
      parameters:
        - in: body
          name: json
          description: The user signup details
          required: true
          schema:
            $ref: '#/definitions/UserSignup'
      responses:
        200:
          description: newly created user
          schema:
            properties:
              status:
                description: "status of the response"
                type: string
                enum: ['OK','Error']
              result:
                $ref: '#/definitions/UserBasic'
```

The UI is parsing it correctly, in that the UI displays the correct fields, and in the request preview it looks correct:

```
POST http://localhost:8080/api/noauth/v1/signup HTTP/1.1
Host: localhost
Accept: application/json
Accept-Encoding: gzip,deflate,sdch
Accept-Language: en-US,en;q=0.8,fa;q=0.6,sv;q=0.4
Cache-Control: no-cache
Connection: keep-alive
Origin: http://localhost:8081
Referer: http://localhost:8081/
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.134 Safari/537.36
Content-Length: 101
Content-Type: application/json
{
  "firstName": "Hank",
  "lastName": "Zill",
  "email": "ha...@example.com",
  "password": "hank1"
}
```

However, when I execute the command, it's sent as a GET. Here is the Chrome console output:
```
Remote Address:127.0.0.1:8080
Request URL:http://localhost:8080/api/noauth/v1/signup?{%20%20%22firstName%22:%20%22Hank%22,%20%20%22lastName%22:%20%22Zill%22,%20%20%22email%22:%20%22hank@example.com%22,%20%20%22password%22:%20%22hank1%22}
Request Method:GET
Status Code:405 Method Not Allowed
```
I get the same error in Firefox. I assume there is something wrong with my definition, but I can't figure out what it is. I have tried setting the 'name' field in the parameters to 'body', but that doesn't help. I'm on Linux (Ubuntu 14.04), and using a version of the editor cloned from Github on August 1st.

_Copied from original issue: swagger-api/swagger.io#27_

**Comment by mohsen1** [comment]
@hzill I can't reproduce this error. It's very likely the error is on your server side. We have end to end tests that test POST functionality with our "Petstore on Heroku" example. Please try the heroku-pets example POST requests and let me know if it's still sending GET requests. Also, look at browser network tab to see outgoing XHR calls to make sure browser is sending the right call.

**Comment by davisford** [comment]
I'm having this same problem.

**Comment by mohsen1** [comment]
@davisford can you see your preflight call and make sure POST is being returned by your server? 

Try the heroku petstore in online editor and look at the preflights call.It returns POST,GET

**Comment by davisford** [comment]
@mohsen1 I don't see any pre-flight requests AFAICT -- what does the preflight call look like?

However, I did drop back to the 2.9.6 release tag instead of building from HEAD, and that seems to have fixed it?

Now, it posts, with the correct payload (the request failed with a 400, but that is my problem)

![screen shot 2015-08-03 at 2 42 15 pm](https://cloud.githubusercontent.com/assets/186677/9044939/02be5c6a-39ee-11e5-85ea-4e10612e69e0.png)

**Comment by mohsen1** [comment]
The preflight call is sent by jQuery. It's a OPTIONS call to make sure the server is capable of accepting a POST call. From 2.9.6 we've updated the versions of Angular and jQuery. It can be an issue with new versions of those dependencies. I'll investigate.

**Comment by davisford** [comment]
I did not see an `OPTIONS` request being sent out.

Incidentally, with the editor, is there no way to set the server's base url for the "Try This" feature?  Currently, I have to host the build statically in my own app so I can try it out against my own api endpoints.  I'm aware of CORS, and could run the browser with security disabled, but it would be nice to be able to launch the build locally from the filesystem, and point it to my running server.  I could not figure out how to achieve from skimming docs.  I'm sure I could hack it into the code, but just wondered if I was missing something.

**Comment by mohsen1** [comment]
Are you aware of `host` property in Swagger? You can point to a different host with that.

**Comment by davisford** [comment]
I'm not sure how I missed that :)  Last (quick) question: when I dropped back to 2.9.6, it seems to be ignoring the `readOnly` property I have on the definitions.  With the latest release, it would only post those values in the body that were not listed as `readOnly` -- was that added just recently?

**Comment by mohsen1** [comment]
Yes, that's a bug in 2.9.6. It's fixed now. #529

**Comment by davisford** [comment]
I seemed to have gotten past most of my issues -- thank your for the help.  I'm building and running the editor on 9000 in dev mode, and pointing `host` to my server, so `readOnly` works again, and I do now see the pre-flight `OPTIONS` requests being sent.

The editor is running `HEAD` but I do notice this error fills up the console - is that expected?

![screen shot 2015-08-03 at 3 36 54 pm](https://cloud.githubusercontent.com/assets/186677/9046005/a78de79a-39f5-11e5-9067-39ef8c369000.png)

**Comment by davisford** [comment]
Although the pre-flight OPTIONS request is url-encoding the payload, is that expected?  

My server responds to the CORS request -- it may be I haven't totally configured it correctly on my side yet, but I do notice that the editor does not issue a `POST` after `OPTIONS` returns 200.

![screen shot 2015-08-03 at 3 42 35 pm](https://cloud.githubusercontent.com/assets/186677/9046106/61ff0dde-39f6-11e5-95f4-098a77f11a16.png)

**Comment by mohsen1** [comment]
Ah I think I know what's going on....

Try `try-method-fix` branch and let me know if that fixes the issue for you.

**Comment by davisford** [comment]
Success -- the angular errors went away and it is now posting the body as expected.

**Comment by hzill** [comment]
I just pulled down that branch and had a successful test with a POST as
well.

Thanks!

**Comment by mohsen1** [comment]
@davisford the error you're seeing should be fixed with 70d4285
